### PR TITLE
[coingecko-api@1.0.9] Add types

### PR DESCRIPTION
**Pull request by janklimo**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ x Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/coingecko-api/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "esModuleInterop": true,
```
Comment (andrewbranch) [inline types/coingecko-api/tsconfig.json:7]:
```suggestion
```

File: types/coingecko-api/coingecko-api-tests.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import CoinGecko from "coingecko-api";
```
Comment (andrewbranch) [inline types/coingecko-api/coingecko-api-tests.ts:1]:
```suggestion
import CoinGecko = require("coingecko-api");
```

**Review by andrewbranch — CHANGES_REQUESTED** [review]
Just browsing through [the docs](https://www.npmjs.com/package/coingecko-api), it looks like there’s quite a lot missing from these types. Incomplete types aren’t necessarily a blocker, but is this enough to be really useful?

**Comment by janklimo** [comment]
@andrewbranch thank you for a quick review! You're certainly right this doesn't cover the whole API. However, it covers the use cases we have in our apps so I'd argue this already adds value. While I won't have time to add more, I'm hoping this gets the ball rolling and enables others to add more types easily.

**Review by peterblazejewicz — APPROVED** [review]
LGMT!
@janklimo thx!

**Review by andrewbranch — APPROVED** [review]
Works for me 👍

**Comment by janklimo** [comment]
Ready to merge

**Pull request merged**